src/test/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParserV2gTest.java

```diff
@@ -0,0 +1,426 @@
+        Assert.assertFalse(record.getId().isEmpty());
+
+        Assert.assertEquals(DateTime.parse("2014-03-24T12:15:41.010Z"), record.getTime());
+
+        final Map<String, ? extends Metric> variables = record.getMetrics();
+        Assert.assertThat(variables, Matchers.hasKey("t1"));
+        final Metric t1 = variables.get("t1");
+        Assert.assertEquals(4, t1.getValues().size());
+        assertValue(t1.getValues().get(0), 1d, Unit.SECOND);
+        assertValue(t1.getValues().get(1), 2d, Unit.MILLISECOND);
+        assertValue(t1.getValues().get(2), 0d, Unit.MILLISECOND);
+        assertValue(t1.getValues().get(3), 4d, Unit.SECOND);
+
+        Assert.assertThat(variables, Matchers.hasKey("t2"));
+        final Metric t2 = variables.get("t2");
+        Assert.assertEquals(3, t2.getValues().size());
+        assertValue(t2.getValues().get(0), 5d);
+        assertValue(t2.getValues().get(1), 6d);
+        assertValue(t2.getValues().get(2), 4d);
+
+        Assert.assertThat(variables, Matchers.hasKey("g1"));
+        final Metric g1 = variables.get("g1");
+        Assert.assertEquals(3, g1.getValues().size());
+        assertValue(g1.getValues().get(0), 1.2d, Unit.GIGABYTE);
+        assertValue(g1.getValues().get(1), 1.1d);
+        assertValue(g1.getValues().get(2), 0.8d, Unit.SECOND);
+
+        Assert.assertThat(variables, Matchers.hasKey("c1"));
+        final Metric c1 = variables.get("c1");
+        Assert.assertEquals(1, c1.getValues().size());
+        assertValue(c1.getValues().get(0), 1d);
+    }
+
+    @Test
+    public void testEmpty() throws ParsingException, IOException {
+        final Record record = parseRecord("QueryLogParserV2gTest/testEmpty.json");
+        Assert.assertNotNull(record);
+
+        Assert.assertEquals(DateTime.parse("2014-03-24T12:15:41.010Z"), record.getTime());
+        Assert.assertEquals(3, record.getAnnotations().size());
+        Assert.assertEquals("MyHost", record.getAnnotations().get(Key.HOST_DIMENSION_KEY));
+        Assert.assertEquals("MyService", record.getAnnotations().get(Key.SERVICE_DIMENSION_KEY));
+        Assert.assertEquals("MyCluster", record.getAnnotations().get(Key.CLUSTER_DIMENSION_KEY));
+        Assert.assertTrue(record.getMetrics().isEmpty());
+    }
+
+    @Test 
+    public void testPresentDimensions() throws ParsingException, IOException {
+        final Record record = parseRecord("QueryLogParserV2gTest/testPresentDimensions.json");
+        Assert.assertNotNull(record);
+
+        Assert.assertNotNull(record.getDimensions());
+        Assert.assertEquals(2, record.getDimensions().size());
+        Assert.assertEquals("DimVal1", record.getDimensions().get("Dim1"));
+        Assert.assertEquals("DimVal2", record.getDimensions().get("Dim2"));
+    }
+
+    @Test
+    public void testMissingDimensions() throws ParsingException, IOException {
+        parseRecord("QueryLogParserV2gTest/testMissingDimensions.json");
```

Comment:
What exactly does this test? -- The above two tests (excluding the new one) test a complete record and a minimal record. This seems unnecessary and without any real assertions.